src/macaron/config/defaults.ini

```diff
@@ -295,6 +295,20 @@ deploy_arg =
 [builder.poetry.ci.deploy]
 github_actions = pypa/gh-action-pypi-publish
 
+# This is the spec for trusted Docker build tool usages.
+[builder.docker]
+entry_conf =
+# Also account for if there's multiple dockerfiles, e.g. test.Dockerfile
+build_configs =
+    Dockerfile
+    Dockerfile.*
+    *.Dockerfile
```

Comment:
Thinking more about this pattern. Have you ever encountered such usage in any of the Github projects? Because the usages like `dev.Dockerfile` or `test.Dockerfile` are very unusual to me. I haven't seen them any where. 
If there is an existing project that use this pattern, it's good to share the link here so that we could discuss too.

Reply:
I chose to include those since while during research I saw somewhere that those would be considered valid names, as such I thought it would be good to include for the sake of completeness. Searching github does show some repos that use this convention though:
- `dev.Dockerfile` in https://github.com/logspace-ai/langflow
- Didn't find one with `test.Dockerfile` specifically from a quick search but did see a few along the lines of `latest.Dockerfile`/similar, e.g. https://github.com/cube-js/cube/blob/c8ef602495edfd6e6c8b898bb9a61bb5be6602ae/packages/cubejs-docker/latest.Dockerfile#L4 among others

Reply:
I think it's also interesting to see how each of those Dockerfile are used and whether it's actually used **build** the final package or not. 
- In the first case - `dev.Dockerfile in https://github.com/logspace-ai/langflow`. I have a quick look and it seems that this is only used for development purposes and doesn't build the final production image. Let's say, a repository has `dev.Dockerfile` for development purposes only and there is not other files that match these patterns, Macaron still consider that repository has Dockerfile build tool (in this case that is a False Positive). Could we do anything to prevent such cases :thinking: ? 
- As for the second case, that is a use case for building the production image.

Reply:
Hm good point, don't know if its a bit much but maybe seeing what gets referenced in pipelines could be an indicator for whether to count a dockerfile or not? Could mean that repos without pipelines could potentially get missed out on, but at least it would be reduced false positives?

Reply:
At this point we can detect candidate build tools. The `build_as_code` check then looks for commands executing the build tool in shell scripts that are reachable from GitHub Actions workflows. I think at that time we are in a better position to report whether the build tool is actually used for build and deployment.